### PR TITLE
Add support for ExternalSecret and SecretStore resources

### PR DESCRIPTION
**Pull request by JGiola**
## What this PR is for?

This PR reintroduce the support for ExternalSecret and SecretStore resources with custom ordering and wait functions.

### CODE REVIEW

File: pkg/extensions/externalsecretmutator.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+// annotateExternalSecret mutate the ExternalSecret obj with the depends-on annotation with all the stores
+// referrend inside it.
+func (m *externalSecretsMutator) annotateExternalSecret(obj *unstructured.Unstructured) error {
+	extsec := new(extsecv1beta1.ExternalSecret)
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, extsec); err != nil {
+		return err
+	}
+	stores := make(sets.Set[resource.ObjectMetadata])
+
+	// check if there is a default secret store set for all external secert
```
Comment (davidebianchi) [inline pkg/extensions/externalsecretmutator.go:146]:
```suggestion
	// check if there is a default secret store set for all external secret
```

File: pkg/extensions/externalsecretmutator_test.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+	externalSecretMap := map[string]*unstructured.Unstructured{
+		"key": jpltesting.UnstructuredFromFile(t, filepath.Join(testdata, "deployment.yaml")),
+	}
+
+	tests := map[string]struct {
+		externalSecretMap map[string]*unstructured.Unstructured
+		obj               *metav1.PartialObjectMetadata
+		expectedResult    bool
+	}{
+		"empty external secret map return alway false": {
```
Comment (davidebianchi) [inline pkg/extensions/externalsecretmutator_test.go:100]:
```suggestion
		"empty external secret map always returns false": {
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,8 +7,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
-- fixed `delete-before-apply` annotation behaviour to call namespace-scoped API
-- update to go 1.22.4
```
[the remark this answers is not in the document]
Comment (JGiola) [reply CHANGELOG.md:11]: They are old changelog never released that are not relevant anymore for this new version

[the remark this answers is not in the document]
Comment (JGiola) [reply CHANGELOG.md:11]: The go version was a mistake, I will add that back with the new patch version

**Review by davidebianchi — APPROVED** [review]
LGTM

**Pull request merged**